test/test_boundary_conditions.jl

```diff
@@ -21,7 +21,7 @@ end
     @testset "Boundary condition instantiation" begin
         @info "  Testing boundary condition instantiation..."
 
-        for C in (Value, Gradient, Flux)
+        for C in (Value(), Gradient(), Flux())
```

Comment:
What's the reason for making this change from uninstantiated to instantiated in this PR? I agree that it represents an improvement, main downside is that it introduces a lot of noise into the review / changes to look over.

Reply:
This was the simplest way I could come up with to pass the `Open` property. In my last commit I added this method:
https://github.com/CliMA/Oceananigans.jl/blob/6a7c91e17398e96950266e17fb17d46375f14c8d/src/BoundaryConditions/boundary_condition.jl#L59-L60
as an alternative to updating all of the tests but I wasn't sure if its better to update them all now?